src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutes.scala

```diff
@@ -0,0 +1,93 @@
+package org.broadinstitute.dsde.workbench.leonardo.api
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.ws._
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import akka.stream.Materializer
+import akka.stream.scaladsl._
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.collection.immutable
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Created by rtitle on 8/4/17.
+  */
+trait ProxyRoutes { self: LazyLogging =>
+  implicit val system: ActorSystem
+  implicit val materializer: Materializer
+  implicit val executionContext: ExecutionContext
+
+  val jupyterPort = 8000
+
+  val proxyRoutes: Route =
+    pathPrefix(Segment) { notebookId =>  // TODO notebookId not currently used
+      extractRequest { request =>
+        complete {
+          request.header[UpgradeToWebSocket] match {
+            case Some(upgrade) => handleWebSocketRequest(request, upgrade)
+            case None => handleHttpRequest(request)
```

Comment:
please break these functions out into a `ProxyService`. see `LeoRoutes` -> `LeonardoService` for pattern